lib/manager/ansible-galaxy/collections.ts

```diff
@@ -0,0 +1,161 @@
+    }
+  }
+  return localDependency;
+}
+
+function handleGitDep(
+  dep: PackageDependency,
+  nameMatch: RegExpExecArray
+): PackageDependency {
+  /* eslint-disable no-param-reassign */
+  dep.datasource = datasourceGitTags.id;
+
+  if (nameMatch) {
+    // if a github.com repository is referenced use github-tags instead of git-tags
+    if (nameMatch.groups.hostname === 'github.com') {
+      dep.datasource = datasourceGithubTags.id;
+    } else {
+      dep.datasource = datasourceGitTags.id;
+    }
+    // source definition without version appendix
+    const source = nameMatch.groups.source;
+    dep.depName = `${nameMatch.groups.hostname}/${nameMatch.groups.depName}`;
+    // remove leading `git+` from URLs like `git+https://...`
+    dep.lookupName = source.replace(/git\+/, '');
+
+    // if version is declared using version appendix `<source url>,v1.2.0`, use it
+    if (nameMatch.groups.version) {
+      dep.currentValue = nameMatch.groups.version;
+    } else {
+      dep.currentValue = dep.managerData.version;
+    }
+  }
+  return dep;
+  /* eslint-enable no-param-reassign */
+}
+
+function handleGalaxyDep(dep: PackageDependency): PackageDependency {
+  /* eslint-disable no-param-reassign */
+  dep.datasource = datasourceGalaxyCollection.id;
+  dep.depName = dep.managerData.name;
+  dep.registryUrls = dep.managerData.source ? [dep.managerData.source] : [];
+  dep.currentValue = dep.managerData.version;
+  return dep;
+  /* eslint-enable no-param-reassign */
+}
+
+function finalize(dependency: PackageDependency): boolean {
+  const dep = dependency;
+  dep.depName = dep.managerData.name;
+
+  const name = dep.managerData.name;
+  const nameMatchRegex = /^(?<source>(git|http|git\+http|ssh)s?(:\/\/|@)(?<hostname>.*)(\/|:)(?<depName>.+\/[^.,]+)\/?(\.git)?)(,(?<version>.*))?$/;
+  const nameMatch = nameMatchRegex.exec(name);
+
+  // use type if defined
+  switch (dependency.managerData.type) {
+    case 'galaxy':
+      handleGalaxyDep(dep);
+      break;
+    case 'git':
+      handleGitDep(dep, nameMatch);
+      break;
+    case 'file':
+      dep.skipReason = SkipReason.LocalDependency;
+      break;
+    case null:
+      // try to find out type based on source
+      if (nameMatch) {
+        handleGitDep(dep, nameMatch);
+        break;
+      }
+      if (galaxyDepRegex.exec(dep.managerData.name)) {
+        dep.datasource = datasourceGalaxyCollection.id;
+        dep.depName = dep.managerData.name;
+        break;
+      }
+      dep.skipReason = SkipReason.NoSourceMatch;
+      break;
+    default:
+      dep.skipReason = SkipReason.Unsupported;
+      return true;
+  }
+
+  if (dependency.currentValue == null && dep.skipReason == null) {
+    dep.skipReason = SkipReason.NoVersion;
+  }
+  return true;
+}
+
+export function extractCollections(lines: string[]): PackageDependency[] {
+  const deps: PackageDependency[] = [];
+
+  for (let lineNumber = 0; lineNumber < lines.length; lineNumber += 1) {
```

Comment:
Isn't it easier to parse yaml with `js-yaml` to object instead of using regexes?

Reply:
If I remember correctly this has been used before in another manager but has been rolled back because of some reason. Are the problems fixed?

Reply:
I'm not sure which problems had occured. @rarkins Do you remember something?